### DISCUSSION TITLE
Get Tenant by subdomain

### DISCUSSION THREAD

**Discussion by alex-rogers-876** [abpframework/abp#6681] [Q&A]
So I'm attempting to use subdomains using the tenant's name to resolve such. In my case, working in local host, I edited my host file to point to a new tenant named subdomains to the localhost application.

I'm able to go-to the page and it loads properly using test.mydomain.com:[port] but it does not seem to honour the domaintenantresolver I set up in ConfigureServices. Is this a completely wrong approach to test subdomain multitenancy?

The code below is the function I added in which I call in ConfigureServices. I also added the following lines to my host file: 127.0.0.1 www.mydomain.com 127.0.0.1 www.meme.mydomain.com

```
private void ConfigureTenantResolver(ServiceConfigurationContext context, IConfiguration configuration)
    {
      
        Configure<AbpTenantResolveOptions>(options =>
        {
            options.TenantResolvers.Insert(1, new DomainTenantResolveContributor($"{0}.mydomain.com:44327"));
            // OR
            // options.AddDomainTenantResolver($"{0}.mydomain.com:44327");
    
          
            options.TenantResolvers.Add(new QueryStringTenantResolveContributor());
            options.TenantResolvers.Add(new RouteTenantResolveContributor());
            options.TenantResolvers.Add(new HeaderTenantResolveContributor());
            options.TenantResolvers.Add(new CookieTenantResolveContributor());
        });
    }
```

**Comment by antosubash** [comment]
There is an issue regarding this which is still open. https://github.com/abpframework/abp/issues/4708

**Comment by maliming** [comment]
hi @alex-rogers-876 

It seems there is a bug with `DomainTenantResolveContributor` I will fix that.

https://github.com/abpframework/abp/pull/7125
